virtool/labels/data.py

```diff
@@ -0,0 +1,140 @@
+from sqlalchemy.exc import IntegrityError
+from typing import List
+
+from aiohttp.web_exceptions import HTTPBadRequest
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy import select
+
+from virtool.data.errors import ResourceConflictError, ResourceNotFoundError
+from virtool.labels.db import SampleCountTransform
+from virtool.labels.oas import EditLabelSchema
+from virtool.mongo.core import DB
+from virtool_core.models.label import Label, LabelMinimal
+from virtool.labels.models import Label as LabelSQL
+from virtool.mongo.transforms import apply_transforms
+from virtool.pg.utils import get_generic
+from virtool.utils import base_processor
+
+
+class LabelsData:
+    def __init__(self, db, pg):
+        self._db: DB = db
+        self._pg = pg
+
+    async def find(self, term: str) -> List[LabelMinimal]:
+        """
+        List all sample labels.
+
+        :param term: the query term
+        :return: a list of all sample labels.
+        """
+        statement = select(LabelSQL).order_by(LabelSQL.name)
+        if term:
+            statement = statement.filter(LabelSQL.name.ilike(f"%{term}%"))
+
+        labels = await get_generic(self._pg, statement)
+
+        documents = await apply_transforms(
+            [label.to_dict() for label in labels], [SampleCountTransform(self._db)]
+        )
+
+        return[
+            LabelMinimal(**label)
+            for label in documents
+        ]
+
+    async def create(self, label_name: str, label_color: str, label_description: str) -> Label:
```

Comment:
Not necessary to have ``label_`` prefix. It is used for `label_id` only to avoid conflict with the built-in `id`.